### ISSUE TITLE
DateTimePicker does not show the week numbers

### ISSUE THREAD

**Issue by McPringle** [vaadin/flow-components#1792]
### Description

The `DateTimePicker`does not show the week number when configured to do so.

### Expected outcome

I would expect to see the week number in the calendar component.

### Actual outcome

The calendar component does not show the week number.

### Live Demo (optional)

![image](https://user-images.githubusercontent.com/1254039/121856236-90e5d500-ccf4-11eb-9d0b-1007086affb9.png)

### Minimal reproducible example

https://github.com/McPringle/vaadin-datetimepicker-weeknumber-bug

### Steps to reproduce

1. Clone the example project
2. Run the example project
3. Open the main page in your browser (http://localhost:8080)
4. Click on the date component to show the calendar

OR

1. Create an instance of the `DateTimePicker` component (example project, file `TestView.java`, line 13)
2. Configure the `DateTimePicker` instance to show the week number (example project, file `TestView.java`, line 14)
3. Run the project
4. Open the main page in your browser (http://localhost:8080)
5. Click on the date component to show the calendar

### Environment

- Vaadin 20.0.1
- macOS 11.4 Big Sur

```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)
```

### Browsers Affected
- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge
- [ ] IE 11
- [x] iOS Safari
- [ ] Android Chrome

I did only check the first three browsers and iOS. I don't have Edge or IE 11 on my MacBook and I don't have an Android phone.

**Comment by rolfsmeds** [comment]
Hi, as mentioned in the javadocs (https://vaadin.com/api/platform/21.0.0.alpha4/com/vaadin/flow/component/datepicker/DatePicker.html#setWeekNumbersVisible-boolean-) and the new component docs (https://vaadin.com/docs/latest/ds/components/date-picker) week numbers are only supported when the first day of the week is set to Monday. This is because the feature is based on ISO week numbers which again are based on Monday being the first day of the week (https://en.wikipedia.org/wiki/Week#Week_numbering).

**Comment by McPringle** [comment]
Thank you very much for pointing this out. I did not expect this behavior. I see calendars showing week numbers even in countries with Sunday as the first day of the week. Nevertheless, I should create a new issue because my system (where I ran my app locally) is configured to use Monday as the first day of the week, so I should see the week numbers…

**Comment by rolfsmeds** [comment]
You may also need to explicitly set the first day of week through the i18n object, `DatePickerI18n setFirstDayOfWeek(1)`, as at least in the screenshots provided it does seem to be Sunday.

**Comment by McPringle** [comment]
Thanks for your additional information. I switched to `EnhancedDatePicker` because it supports setting the date format pattern. It had the same problem with week numbers and week start date. So I created a [simple provider class](https://github.com/komunumo/komunumo-server/blob/main/src/main/java/org/komunumo/util/LocalizedEnhancedDatePickerI18NProvider.java) to automatically generate the `I18n` object based on the locale.
